### PR TITLE
Replace quarkus PGVectorEmbeddingStore with new Langchain4J PGVectorEmbeddingStore

### PR DESCRIPTION
**Pull request by humcqc**
Replace quarkus PGVectorEmbeddingStore with new Langchain4J PGVectorEmbeddingStore

Allow metadata filtering and remove features

@jmartisk @sebastienblanc @geoand @langchain4j this PR replaces https://github.com/quarkiverse/quarkus-langchain4j/pull/410

### CODE REVIEW

File: embedding-stores/pgvector/runtime/src/main/java/io/quarkiverse/langchain4j/pgvector/runtime/PgVectorEmbeddingStoreRecorder.java [hunk 1]
```diff
@@ -15,19 +18,17 @@ public class PgVectorEmbeddingStoreRecorder {
...
-                return new PgVectorEmbeddingStore(dataSource, config.table(), config.dimension(), config.useIndex(),
-                        config.indexListSize(), config.createTable(), config.dropTableFirst());
-            }
+        return context -> {
+            AgroalDataSource dataSource = Optional.ofNullable(datasourceName)
+                    .map(DataSourceLiteral::new)
+                    .map(dl -> context.getInjectedReference(AgroalDataSource.class, dl))
+                    .orElse(context.getInjectedReference(AgroalDataSource.class, new Default.Literal()));
+
+            dataSource.flush(AgroalDataSource.FlushMode.GRACEFUL);
```
Comment (humcqc) [inline embedding-stores/pgvector/runtime/src/main/java/io/quarkiverse/langchain4j/pgvector/runtime/PgVectorEmbeddingStoreRecorder.java:27]: The first build [failed](https://github.com/quarkiverse/quarkus-langchain4j/actions/runs/9214412003/job/25350530831), i think it's link to the fact we are using FlushMode.GRACEFUL, perhaps FlushMode.ALL will be better. WDYT ?

**Comment by geoand** [comment]
Thanks for this!

I would like to do the update to 0.31 in #622, so once that is in, please rebase this PR onto it.

**Comment by geoand** [comment]
https://github.com/quarkiverse/quarkus-langchain4j/pull/622 has been merged

**Comment by jmartisk** [comment]
Sorry for the delay, traveling all week, but I'll give it a shot as soon as I can

**Review by jmartisk — APPROVED** [review]
Ok, tried it and it works.
Super awesome, thanks!

**Pull request merged**